src/components/SecondaryPanes/Frames/index.js

```diff
@@ -201,27 +200,32 @@ function appendSource(sources, frame) {
   });
 }
 
-const getAndProcessFrames = createSelector(
-  getFrames,
-  getSources,
-  (frames, sources) => {
-    if (!frames) {
-      return null;
-    }
+export function getAndProcessFrames(frames, sources) {
+  if (!frames) {
+    return null;
+  }
 
-    frames = frames
-      .filter(frame => getSourceForFrame(sources, frame))
-      .filter(frame => !get(frame, "source.isBlackBoxed"))
-      .map(frame => appendSource(sources, frame))
-      .map(annotateFrame);
+  const processedFrames = frames
+    .filter(frame => getSourceForFrame(sources, frame))
+    .map(frame => appendSource(sources, frame))
+    .filter(frame => !get(frame, "source.isBlackBoxed"))
+    .map(annotateFrame);
```

Comment:
the issue was that we were filtering before we had sources to filter out the blackbox data from